File: app/src/main/java/no/joharei/flixr/mainpage/MainApi.java
package no.joharei.flixr.mainpage;

import javax.inject.Inject;

import no.joharei.flixr.api.FlickrApiContainer;
import no.joharei.flixr.api.models.Photosets;
import no.joharei.flixr.mainpage.models.Contacts;
import rx.Observable;

public class MainApi {
    private FlickrApiContainer flickrApiContainer;

    @Inject
    MainApi(FlickrApiContainer flickrApiContainer) {
        this.flickrApiContainer = flickrApiContainer;
    }

    Observable<Photosets> getMyPhotosets() {
        return flickrApiContainer.getPhotosets(null);
    }

    Observable<Contacts> getContacts() {
        return flickrApiContainer.getContacts();
    }
}

File: app/src/main/java/no/joharei/flixr/mainpage/MainApi.kt
package no.joharei.flixr.mainpage

import io.reactivex.Flowable
import io.reactivex.Observable
import no.joharei.flixr.api.FlickrApiContainer
import no.joharei.flixr.api.models.Photosets
import no.joharei.flixr.mainpage.models.Contacts
import no.joharei.flixr.tools.ObservableCache
import javax.inject.Inject

class MainApi
@Inject
constructor(private val flickrApiContainer: FlickrApiContainer, private val observableCache: ObservableCache) {

    fun fetchMyPhotosets(): Flowable<Photosets> = flickrApiContainer.getPhotosets(null)

    fun fetchContacts(): Observable<Contacts> {
        return observableCache.getCachedObservable(flickrApiContainer.getContacts(), Contacts::class.java, true, true)
    }

    fun clearCache() = observableCache.clearCache()
}
